Tribler/Core/Upgrade/db72_to_pony.py

```diff
@@ -263,6 +262,25 @@ def convert_personal_channel(self):
 
         v.value = CONVERSION_FINISHED
 
+    @inlineCallbacks
+    def update_convert_total(self, amount, elapsed):
+        if self.notifier_callback:
+            self.notifier_callback("%i entries converted in %i seconds (%i e/s)" % (amount, int(elapsed),
+                                                                                    int(amount / elapsed)))
+            yield deferLater(reactor, 0.001, lambda: None)
```

Comment:
I assume that the notifier callback invokes the event endpoint and passes a message to the GUI? If so, I think sending notifications to the GUI at millisecond speed is too fast. We might want to send a message to the GUI e.g. every half a second?

Please correct me if I'm wrong.

Reply:
This is reactor magic. The only reason why this `deferLater` is here is to give other Deferreds running on the same reactor a chance to do their stuff in between `notifier_callback` calls. This thing only truly happens once per conversion loop iteration (a single batch), that is *about 1 second*.  The whole thing works like this:
"at the end of converting each batch (that takes approx 1 second), give control to the reactor and schedule `notifier_callback` to be run in 1 millisecond."

Reply:
Ah I see, missed that 👍